simulator_main/services/BitstreamService.cs

```diff
@@ -2,55 +2,63 @@
         {
-            this.IcdDictionary = new Dictionary<string, Type>();
-            IcdDictionary.Add("FlightBoxDownIcd", typeof(FlightBoxIcd));
-            IcdDictionary.Add("FlightBoxUpIcd", typeof(FlightBoxIcd));
-            IcdDictionary.Add("FiberBoxDownIcd", typeof(FiberBoxIcd));
-            IcdDictionary.Add("FiberBoxUpIcd", typeof(FiberBoxIcd));
+            TelemetryConnection = telemetryConnection;
+            this.IcdDictionary = new Dictionary<string, (Type, IcdTypes)>();
+            IcdDictionary.Add("FlightBoxDownIcd", (typeof(FlightBoxIcd), IcdTypes.FlightBoxDown));
+            IcdDictionary.Add("FlightBoxUpIcd", (typeof(FlightBoxIcd), IcdTypes.FlightBoxUp));
+            IcdDictionary.Add("FiberBoxDownIcd", (typeof(FiberBoxIcd), IcdTypes.FiberBoxDown));
+            IcdDictionary.Add("FiberBoxUpIcd", (typeof(FiberBoxIcd), IcdTypes.FiberBoxUp));
         }
-        public async Task<string> GetPacketDataAsync(GetSimulationDto getSimulationDto)
+        public async Task ConnectTelemetry()
+        {
+            await TelemetryConnection.ConnectAsync();
+        }
+        public async Task GetPacketDataAsync(GetSimulationDto getSimulationDto)
         {
             if (!IcdDictionary.ContainsKey(getSimulationDto.IcdName))
-                return "400";
+                return;
 
             string jsonText = File.ReadAllText(ICD_REPO_PATH + getSimulationDto.IcdName + ICD_FILE_TYPE);
 
-            // get the icd generic type at run time
-            Type genericIcdType = typeof(IcdPacketGenerator<>).MakeGenericType(IcdDictionary[getSimulationDto.IcdName]);
-            // construct the generic icd at runtime
-            dynamic icdInstance = Activator.CreateInstance(genericIcdType);
 
-            return await icdInstance.GeneratePacketBitData(jsonText,0,0);
+            Type genericIcdType = typeof(IcdPacketGenerator<>).MakeGenericType(IcdDictionary[getSimulationDto.IcdName].Item1);
 
+            dynamic icdInstance = Activator.CreateInstance(genericIcdType);
+
+            byte[] packetValue = await icdInstance.GeneratePacketBitData(jsonText, 0, 0);
+            await TelemetryConnection.SendPacket(packetValue, IcdDictionary[getSimulationDto.IcdName].Item2);
         }
-        public async Task<string> GetPacketErrorDataAsync(GetErrorSimulationDto getSimulationErroDto)
+        public async Task GetPacketErrorDataAsync(GetErrorSimulationDto getSimulationErroDto)
         {
             if (!IcdDictionary.ContainsKey(getSimulationErroDto.IcdName))
-                return "400";
-            
+                return;
+
             string jsonText = File.ReadAllText(ICD_REPO_PATH + getSimulationErroDto.IcdName + ICD_FILE_TYPE);
 
-            Type genericIcdType = typeof(IcdPacketGenerator<>).MakeGenericType(IcdDictionary[getSimulationErroDto.IcdName]);
+            Type genericIcdType = typeof(IcdPacketGenerator<>).MakeGenericType(IcdDictionary[getSimulationErroDto.IcdName].Item1);
```

Comment:
why add Item1?

Reply:
Change that key of dictionary will be name of IcdType and key will be the type like this: <IcdType, Type>